web/src/engine/osk/src/banner/bannerController.ts

```diff
@@ -2,134 +2,84 @@ import { DeviceSpec } from '@keymanapp/web-utils';
 import type { PredictionContext, StateChangeEnum } from '@keymanapp/input-processor';
 import { ImageBanner } from './imageBanner.js';
 import { SuggestionBanner } from './suggestionBanner.js';
-import { BannerView, BannerOptions, BannerType } from './bannerView.js';
+import { BannerView } from './bannerView.js';
 import { Banner } from './banner.js';
 import { BlankBanner } from './blankBanner.js';
+import { HTMLBanner } from './htmlBanner.js';
 
 export class BannerController {
-  private _activeType: BannerType;
-  private _options: BannerOptions = {};
   private container: BannerView;
-  private alwaysShow: boolean;
-  private imagePath?: string = "";
 
   private predictionContext?: PredictionContext;
 
   private readonly hostDevice: DeviceSpec;
 
-  public static readonly DEFAULT_OPTIONS: BannerOptions = {
-    alwaysShow: false,
-    imagePath: ""
-  }
+  private _inactiveBanner: Banner;
+
+  /**
+   * Builds a banner for use when predictions are not active, supporting a single image.
+   */
+  public readonly ImageBanner = ImageBanner;
+
+  /**
+   * Builds a banner for use when predictions are not active, supporting a more generalized
+   * content pattern than ImageBanner via `innerHTML` specifications.
+   */
+  public readonly HTMLBanner = HTMLBanner;
 
   constructor(bannerView: BannerView, hostDevice: DeviceSpec, predictionContext?: PredictionContext) {
     // Step 1 - establish the container element.  Must come before this.setOptions.
```

Comment:
`this.setOptions` is now gone?

Reply:
Yeah.  That paradigm started to feel _really messy_ once I started adding in the HTML banner, especially since it obfuscates what properties are needed for which banner.  Direct use of banner constructors is far, far clearer in that regard.